test/core/end2end/fixtures/h2_tls_common.h

```diff
@@ -238,8 +238,8 @@ inline grpc_channel_credentials* create_tls_channel_credentials(
 inline grpc_server_credentials* create_tls_server_credentials(
     fullstack_secure_fixture_data* ffd) {
   grpc_tls_credentials_options* options = grpc_tls_credentials_options_create();
-  options->set_min_tls_version(ffd->tls_version);
-  options->set_max_tls_version(ffd->tls_version);
```

Comment:
The current h2 ssl/tls tests contain a lot of tests with similar code. Taking `h2_ssl_tls12.cc` and `h2_ssl_tls13.cc` for example, most of the code is same, except maybe just one line setting the TLS version.
Would you mind updating them so that they can be better structured?
You can do this in a separate PR, or combining that in this PR. Thank you!

Reply:
Of course not!

But I think those two files you named test the old SSL APIs while my current PR focuses on plumbing the new TLS APIs. Therefore, it's probably better to re-structure those in a separate PR, which I will work on after this one. Let me know if I am missing anything.

Reply:
By taking a closer look at all the fixtures, it seems to me that they fall into different categories, and code is largely duplicated within each of them. For example, the `h2_sockpair...` is one of them. My guess is they are intentionally broken into different test binaries so that they can be executed in parallel. This makes me wonder if a merge of tls12 and tls13 is needed. Let me know what you think. Thanks.

Reply:
I saw you opened another PR...was that intended to solve this particular comment?

Reply:
Yeah that PR was to reduce redundant code by refactoring common functions into h2_tls_common.h. As I pointed out, these tests are all for the old SSL APIs though, so I didn't include the change into this PR.

Reply:
I see, sure I will close this thread as you plan to do that in a separate PR